packages/manager/modules/anthos/src/dashboard/general-information/upgrade-pack/upgrade-pack.routing.js

```diff
@@ -0,0 +1,40 @@
+import angular from 'angular';
+import {
+  pricingConstants,
+  workflowConstants,
+} from '@ovh-ux/manager-product-offers';
+import { TRACKING_PREFIX } from '../../constants';
+
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('anthos.dashboard.general-information.upgrade-pack', {
+    url: '/upgrade-pack',
+    views: {
+      'anthosTenantView@anthos.dashboard': 'anthosDashboardUpgradePack',
+    },
+    params: {
+      pack: { value: null },
+    },
+    resolve: {
+      breadcrumb: () => null,
+      pack: /* @ngInject */ ($transition$) => $transition$.params().pack,
+      pricingType: () => pricingConstants.PRICING_CAPACITIES.UPGRADE,
+      workflowType: () => workflowConstants.WORKFLOW_TYPES.SERVICES,
+      workflowOptions: /* @ngInject */ (packInfo, serviceInfo, atInternet) => ({
+        serviceId: serviceInfo.serviceId,
+        durationToUse: 'P1M',
+        plancodes: angular.copy(packInfo.upgrades),
+        onValidateSubmit() {
```

Comment:
I try to find the usage of this function but nothing.
Is't used ?

Reply:
It is triggered by the `ovh-manager-product-offers` component at the end of the product configuration
```html
    <!-- packages/manager/modules/product-offers/src/product-offers.html L68 -->
    <oui-step-form
        data-on-submit="$ctrl.workflow.onValidateSubmit(validate)"
    >
```